chain/network-primitives/src/types.rs

```diff
@@ -176,12 +178,99 @@ impl KnownPeerState {
     }
 }
 
+// Exists here for testing purposes. Usually implmented by tokio::net::TcpStream
+// but mocked out in tools/mock_node
+pub trait TcpStream: AsyncRead + AsyncWrite + Sync + Send + Unpin {
+    fn local_addr(&self) -> io::Result<SocketAddr>;
+    fn peer_addr(&self) -> io::Result<SocketAddr>;
+}
+
+#[async_trait]
+pub trait TcpListener {
+    async fn accept(&self) -> io::Result<(Box<dyn TcpStream>, SocketAddr)>;
+}
```

Comment:
yeah I wonder which is the way to go... I actually tried that first but backed up and just did it with a `Box<dyn TcpStream>` because the generic `<T>` gets propagated to so much stuff that it makes the PR a lot bigger (now we need to change every instance of PeerManager to PeerManager<T> and same with PeerManagerMessageRequest, PeerActor, etc...). So i figured it wont hurt too much to make it dynamic dispatch, as we're already doing that in a few places anyway (lots in struct Client for example). but could go either way

Reply:
I was afraid of that.